src/panels/LogPanel/ScriptLogPanel.tsx

```diff
@@ -31,46 +37,133 @@ export function ScriptLogPanel() {
 
     setScriptLogEntries(
       Object.values(data)
-        .filter((script) => script)
+        .filter((script) => {
+          if (!script.trim()) {
+            return false;
+          }
+          // Cleaning up entries caused by this panel
+          if (script.includes('ScriptLog.txt')) {
+            return false;
+          }
+          // Assuming few people use script scheuduling this should be fine
+          return !script.includes('openspace.scheduleScript');
+        })
         .reverse()
+        .map((script) => {
+          if (script.startsWith('return openspace.')) {
+            return script.substring('return'.length);
+          }
+          return script;
+        })
     );
+
+    setSelectedScripts(new Set());
   }, [luaApi]);
 
   useEffect(() => {
     fetchScriptLogEntries();
   }, [fetchScriptLogEntries]);
 
-  function preprocessScriptEntry(entry: string): string {
-    // @TODO (anden88, 2025-04-17): do we want to rewrite this using regex instead?
-    // return entry.replace(/[.,()""]/g, ' ');
-    return entry
-      .replaceAll('.', ' ')
-      .replaceAll(',', ' ')
-      .replaceAll('"', ' ')
-      .replaceAll('(', ' ')
-      .replaceAll(')', ' ');
+  function sanitizeScript(script: string): string {
+    // Replace special characters with spaces. The '+' ensures repeated punctuation is
+    // replaced by a single space instead of many
+    return script.replace(/[.,()[\]{}"'\\/]+/g, ' ');
+  }
+
+  function handleToggleSelection(index: number) {
+    setSelectedScripts((prev) => {
+      const newSet = new Set(prev);
+      if (newSet.has(index)) {
+        newSet.delete(index);
+      } else {
+        newSet.add(index);
+      }
+      return newSet;
+    });
+  }
+
+  function runSelectedScripts() {
+    if (scriptLogEntries === undefined || !luaApi) {
+      return;
+    }
+    // TODO (anden88 2025-04-23): In what order do we want to run the scripts? The scripts
+    // are ordered latest first (top-bottom) but they were actually executed bottom up...
+    const scriptsToRun = Array.from(selectedScripts)
+      .sort((a, b) => a - b)
+      .map((index) => scriptLogEntries[index]);
```

Comment:
Yeah, this is probably a bit confusing for a user. One way to get around the issue is to have a tooltip on the rerun scripts button?

I think we definitely want to run the scripts bottom-up.

Another thing to consider is that now, with the checkboxes, it takes three clicks to re-run two scripts whereas re-running two scripts with the regular button takes two clicks. Just a thought